src/Features/Core/Portable/EventListener/EventListenerTracker.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.EventListener
+{
+    internal class EventListenerTracker<TService>
+    {
+        private readonly HashSet<string> _eventListenerInitialized;
```

Comment:
Add a comment this is the kinds being stashed in here -- I wasn't sure if it was the well-known event kinds or something.

Reply:
sure